lib/types.nix

```diff
@@ -873,8 +873,13 @@ rec {
       description = "function that evaluates to a(n) ${optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType}";
       descriptionClass = "composite";
       check = isFunction;
-      merge = loc: defs:
-        fnArgs: (mergeDefinitions (loc ++ [ "<function body>" ]) elemType (map (fn: { inherit (fn) file; value = fn.value fnArgs; }) defs)).mergedValue;
+      merge = loc: defs: {
+        __functionArgs = lib.pipe defs [
```

Comment:
Could you rephrase this without the `pipe` call?
That will save a number of list allocations and not strictly necessary function calls in the definition of `pipe`.
(Wouldn't matter for individual call sites, but we try to keep the module system as quickly interpreted as it reasonably can.)